Fix related products E2Es for WC9.5

tomalec opened the pull request:
### Changes proposed in this Pull Request:

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

This PR adjusts E2E tests for WC 9.5 changes.

In WC 9.5, by default, there is no related products section on a single product page - I made the test skip in such a case.
Also, I added a "related" tag to all simple products to make sure simple products will appear in the related products section instead of "Cat teasers" - without the "Add to cart" button.



### Screenshots:

![image](https://github.com/user-attachments/assets/7c9a774c-2b82-4421-8184-0c014d0434d2)


<!--- Optional --->


### Detailed test instructions:
<!-- Add detailed instructions for how to test that this PR fixes the issue and confirm that it doesn't break any other features :) -->

1. Run E2E tests locally
2. Run [E2E workflow](https://github.com/woocommerce/google-listings-and-ads/actions/workflows/e2e-tests.yml) with `9.5.0-rc.1` - https://github.com/woocommerce/google-listings-and-ads/actions/runs/12361402622
3. Run [E2E workflow](https://github.com/woocommerce/google-listings-and-ads/actions/workflows/e2e-tests.yml) with `9.4.0` - https://github.com/woocommerce/google-listings-and-ads/actions/runs/12361396232
4. Review the code to check the tests are testing what they should


### Additional details:

<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with change type prefix`(Fix|Add|…) - `, for example:
> Break - A change breaking previous API or functionality.
> Add - A new feature, function or functionality was added.
> Update - Big changes to something that wasn't broken.
> Fix - Took care of something that wasn't working.
> Tweak - Small change, that isn't actually very important.
> Dev - Developer-facing only change.
> Doc - Updated customer or developer facing documentation

If you remove the "Changelog entry" header, the Pull Request title will be used as the changelog entry.

Add the `changelog: none` label if no changelog entry is needed.
-->
### Changelog entry

> Dev - Improve E2E tests for related products.

tomalec committed:
Make E2E's add-to-cart selector more flexible

tomalec committed:
Skip when there is no related prodcuts section

tomalec committed:
Add dedicated test for related products block.

tomalec commented:
See also https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/481

eason9487 commented on tests/e2e/utils/block-page.js:
The type seems to be `Promise<string>`.

eason9487 commented on tests/e2e/utils/block-page.js:
The type seems to be `Promise<string>`.

eason9487 reviewed (approved):
Thanks for the compatibility fix! LGTM.

tomalec committed:
Fix type declaration in jsdoc

Address https://github.com/woocommerce/google-listings-and-ads/pull/2729#pullrequestreview-2507803613
